server.js

```diff
@@ -1,36 +1,161 @@
 
 app.get('/api/v1/foods/:id', function(request, response){
   var id = request.params.id
   Food.find(id)
   .then(function(data){
-    if(data.rowCount == 0) { return response.sendStatus(404)}
+    if(data.rowCount == 0) {
+      return response.sendStatus(404)
+    }
     response.json(data.rows[0])
   })
 })
 
+app.post('/api/v1/foods', function(request, response){
+  // NEEDS HELP! request.query allows for function to work in postman, but fails tests
+  // request.body allows tests to pass, but does not work in postman!
+
+  var name = request.body.name
+  var calories = request.body.calories
+  if(name && calories){
+    Food.createFood(name, calories)
+    .then(function(data){
+      Food.allFoods().then(function(data){
+        return response.status(201).json(data)
+      })
+    })
+  }
+    if(!name){
+      return response.status(422).send({ error: "A food item must have a name"})
+    }
+    if(!calories){
+      return response.status(422).send({ error: "A food item must have calories"})
+    }
+})
+
+app.put('/api/v1/foods/:id', function(request, response){
+  // NEEDS HELP! request.query allows for function to work in postman, but fails tests
+  // request.body allows tests to pass, but does not work in postman!
+  var id = request.params.id
+  var name = request.body.name
+  var calories = request.body.calories
+
+  Food.find(id)
+  .then(function(data){
+    if(data.rowCount == 0) {
+      return response.sendStatus(404)
+    }
+
+    if(name && calories){
+      Food.updateAll(id, name, calories)
+      .then(function(data){
+        Food.allFoods().then(function(data){
+          return response.status(201).json(data)
+        })
+      })
+    }
+
+    if(!calories){
+      Food.updateName(id, name)
+      .then(function(data){
+        Food.allFoods().then(function(data){
+          return response.status(201).json(data)
+        })
+      })
+    }
+
+    if(!name){
+      Food.updateCalories(id, calories)
+      .then(function(data){
+        Food.allFoods().then(function(data){
+          return response.status(201).json(data)
+        })
+      })
+    }
+  })
+})
+
+app.delete('/api/v1/foods/:id', function(request, response){
+  var id = request.params.id
+
+  Food.find(id)
+  .then(function(data){
+    if(data.rowCount == 0) {
+      return response.sendStatus(404)
+    }
+    Food.deleteFood(id)
+    .then(function(data){
+      Food.allFoods().then(function(data){
+        return response.status(200).json(data)
+      })
+    })
+  })
+})
+
+
+// MEALS API ENDPOINTS
+app.get('/api/v1/meals', function(request, response) {
+   Meal.allMeals()
+  .then(function(data){
+    if(data.rowCount == 0){
+      return response.sendStatus(404)
+    }
+    eval(pry.it)
+    response.json(data.rows)
+
+    // allTheMeals.forEach(function(meal){
+    //   allTheMeals['foods'] = Meal.foods(meal.id)
+    // })
+    // .then(function(data){
+    //   return response.status(200).json(data)
+    // })
+    // eval(pry.it)
+
+  })
+})
```

Comment:
Is this what you're asking about in your PR comments? There's a lot in the PR, but this looks like what you were asking about in class.

I think you just need to keep in mind which data you have available to you inside a then. You can always put a `pry.it` inside a then function to see what was passed to it. Also, in your current commented code, I don't think you can `.then()` off of `forEach`, so you want to try a different strategy.